File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> stk;
    int i = 0;
    while(i < arr.size())
    {
        if (!stk.size())
        {
            stk.push_back(arr[i]);
            i++;
        }
        else if (stk.size() && stk.back() == arr[i])
        {
            stk.pop_back();
            i++;
        }
        else if (stk.size() && stk.back() != arr[i])
        {
            stk.push_back(arr[i]);
            i++;
        }
    }
    
    return stk.size() ? stk : vector<int> (1, -1);
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.ArrayDeque;
import java.util.Deque;

class Solution {
    public int[] solution(int[] arr) {
        Deque<Integer> stk = new ArrayDeque<>();
        for (int i = 0; i < arr.length; i++) {
            if (stk.isEmpty()) {
                stk.addLast(arr[i]);
            } else if (stk.peekLast() == arr[i]) {
                stk.pollLast();
            } else {
                stk.addLast(arr[i]);
            }
        }

        if (stk.isEmpty()) return new int[]{-1};
        return stk.stream().mapToInt(i -> i).toArray();
    }
}
